core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -389,6 +389,11 @@ private void failAndRespond(Throwable cause, AggregatedHttpMessage message, Http
                 responseEncoder.writeHeaders(id, streamId, headers, false);
                 future = responseEncoder.writeData(id, streamId, content, true);
             }
+
+            if (!loggedResponseHeadersFirstBytesTransferred) {
```

Comment:
It seems not very different, but how about doing this in the future listener? (like the following code)
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java#L336-L340 
It sets `responseFirstBytesTransferred` only if it succeeded.

Reply:
Thanks, good idea.